scalapact-core/src/main/scala/com/itv/scalapactcore/common/matching/InteractionMatchers.scala

```diff
@@ -13,52 +13,34 @@ import scala.xml._
 
 object InteractionMatchers {
 
-  import StatusMatching._
-  import PathMatching._
-  import HeaderMatching._
-  import MethodMatching._
-  import BodyMatching._
-
   lazy val matchRequest: Boolean => List[Interaction] => InteractionRequest => Either[String, Interaction] = strictMatching => interactions => received =>
-    interactions.find { ir =>
-      if(strictMatching) isStrictRequestMatch(ir)(received)
-      else isNonStrictRequestMatch(ir)(received)
+    interactions.find { interaction =>
+      InteractionMatchingPrograms
+        .matchRequestProgram(interaction.request, received)
```

Comment:
Here was are calling our "program" with it's arguments but it hasn't been run yet because it doesn't know what to do at each stage,